Sources/MongoSwift/BSON/BSONEncoder.swift

```diff
@@ -149,27 +149,38 @@ public class BSONEncoder {
 
         let encoder = _BSONEncoder(options: self.options)
 
-        guard let boxedValue = try encoder.box_(value) else {
-            throw EncodingError.invalidValue(
-                value,
-                EncodingError.Context(
-                    codingPath: [],
-                    debugDescription: "Top-level \(T.self) did not encode any values."
+        do {
+            let optionalBoxedValue = try encoder.box_(value)
+            guard let boxedValue = optionalBoxedValue else {
```

Comment:
I think you can just one line this, `guard let boxedValue = try encoder.box_(value) else { ...`?